jenkins/vm-reset

```diff
@@ -128,7 +133,23 @@ vm_snapshot_revert() {
           )
     >&2 jq -r '.message' <<< "$resp"
     [[ $(jq '.success' <<< "$resp") == true ]] || die 'Cannot revert snapshot'
-    sleep 120  # XXX Can we do better?
+    
+    local snapshot_task_id
+    snapshot_task_id=$(jq -r '.task_id' <<< "$resp")
+
+    local interval=30  # seconds
+    local wait_steps=8 # wait for snapshot revert
+    while true; do
```

Comment:
this while loop and wait_till() look similar. Why don't we use wait_till() here instead of duplicating the code?

Reply:
@mssawant power and task specific actions has different time out and function call to check the status, to keep it simple I made it separate. We can use wait_till() for both power and task but then we need to add more checks, arguments inside it to monitor current operation.

Reply:
Yes, please try passing the required timeout as an argument. I don't see many of them, it would be good to re-use the function.